src/megachatapi_impl.cpp

```diff
@@ -9194,22 +9194,17 @@ void MegaChatContainsMetaPrivate::setTextMessage(const string &text)
     mText = text;
 }
 
-void MegaChatContainsMetaPrivate::setGiphy(MegaChatGiphy* giphy)
+void MegaChatContainsMetaPrivate::setGiphy(std::unique_ptr<MegaChatGiphy>& giphy)
```

Comment:
The idea is to receive the ownership directly, instead of a reference + move. That way, it's clear that the caller gives the ownership (you would need to `setGiphy(move(giphy))` and here you simply do `mGiphy.reset(giphy)` :)

Reply:
made the suggested changes pertaining to ownership. 
I think we can't do a mGiphy.reset(giphy). std::move() seems to be the right operation.